src/ray/raylet/node_manager.cc

```diff
@@ -2957,6 +2968,13 @@ void NodeManager::ProcessSubscribePlasmaReady(
 
   auto message = flatbuffers::GetRoot<protocol::SubscribePlasmaReady>(message_data);
   ObjectID id = from_flatbuf<ObjectID>(*message->object_id());
+
+  // Tell task dependency manager to pull the objects.
+  // NOTE(simon): This doesn't handle cancellation and edge case invovling eviciton.
+  std::vector<rpc::ObjectReference> refs = {FlatbufferToSingleObjectReference(
+      *message->object_id(), *message->owner_address())};
+  task_dependency_manager_.SubscribeWaitDependencies(associated_worker->WorkerId(), refs);
```

Comment:
Just purely out of curiosity, but why do we use `SubscribeWaitDependencies ` instead of `SubscribeGetDependencies `?

Reply:
The methods are not named that well, but I copied the main differences from the header file:
```
SubscribeWaitDependencies
  /// The TaskDependencyManager will track all remote dependencies until the
  /// dependencies are local, or until UnsubscribeWaitDependencies is called
  /// with the same worker ID, whichever occurs first.
```
```
SubscribeGetDependencies
  /// The TaskDependencyManager will track the task's dependencies
  /// until UnsubscribeGetDependencies is called on the same task ID.
```

So SubscribeGetDependencies is supposed to be for tasks that are blocked on some dependencies, and SubscribeWaitDependencies is for a worker that is subscribing asynchronously to stream of dependencies (like with `ray.wait`). I think this class could use a refactor though :)